mbf_abstract_nav/src/move_base_action.cpp

```diff
@@ -540,10 +524,14 @@ void MoveBaseAction::actionRecoveryDone(
       );
       break;
     case actionlib::SimpleClientGoalState::PREEMPTED:
-      ROS_INFO_STREAM_NAMED("move_base",
-                             "The recovery action has been preempted!");
-      if(action_state_ == CANCELED)
-        goal_handle_.setCanceled();
+      ROS_INFO_STREAM_NAMED("move_base", "The action \"" << "recovery" << "\" has been preempted!");
+      if (action_state_ == CANCELED)
```

Comment:
move base has been preempted. otherwise we have preempted internally, (what I don't think can happen) so we don't end the action